internal/provider/resource_gitlab_pages_domain_test.go

```diff
@@ -0,0 +1,124 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/xanzy/go-gitlab"
+)
+
+func TestAccGitlabPagesDomain_basic(t *testing.T) {
+	var pagesDomain gitlab.PagesDomain
+	rInt := acctest.RandInt()
+	testAccCheck(t)
+	project := testAccCreateProject(t)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabPagesDestroy,
+		Steps: []resource.TestStep{
+			// Create a pages domain with all options
```

Comment:
Let's add another test case to only supply the required data - there may be some issues in the current implementation when `pagesDomain.Certificate` is `nil` or even earlier when `go-gitlab` fails, because `nil` is passed to `gitlab.String()` - these are just suspicions though and everything might be okay as-is 👯

Reply:
I removed the `Default` values and changed to testcase to include only the required ones.

Reply:
As per my other comment, let's have multiple test cases: testing w/ and w/o optionals set.